create-infra.yml

```diff
@@ -75,3 +75,100 @@
       ansible.builtin.import_role:
         name: "libvirt_manager"
         tasks_from: "deploy_layout.yml"
+
+    - name: Gather ansible_user_id
+      ansible.builtin.setup:
+        gather_subset:
+          - user_id
+          - user_dir
+
+    - name: Read inventory groups
+      ansible.builtin.find:
+        paths: "{{ ansible_user_dir }}/ci-framework-data/reproducer-inventory/"
+        file_type: file
+        patterns: '*.yml,*.yaml'
+      register: "_all_groups"
+
+    - name: Read groups inventory
+      ansible.builtin.slurp:
+        path: "{{ item.path }}"
+      register: _reproducer_inventory_files
+      loop: "{{ _all_groups.files }}"
+
+    - name: Set groups fact
+      when: "'all' != _group_name"
+      vars:
+        _content: "{{ _inventory_file.content | b64decode | from_yaml }}"
+        _group_name: "{{ _content | list | first }}"
+        _nodes: "{{ _content[_group_name]['hosts'].keys() }}"
+        _group_nodes: >-
+          {%- set group = {} -%}
+          {%- set _ = group.update({_group_name: _nodes}) -%}
+          {{ group }}
+      ansible.builtin.set_fact:
+        _vm_groups: >-
+          {{ _vm_groups | default({}) | combine(_group_nodes) }}
+      loop: "{{ _reproducer_inventory_files.results }}"
+      loop_control:
+        loop_var: "_inventory_file"
+
+    - name: Ensure controllers vms are started and reachable
+      vars:
+        _cifmw_libvirt_manager_layout:
+          vms:
+            controllers:
+              start: true
+              disk_file_name: "dummy"
+        cifmw_libvirt_manager_all_vms: >-
+          {%- set vms = {} -%}
+          {%- for vm in _vm_groups['controllers'] -%}
+            {%- set _ = vms.update({vm: 'controllers'}) -%}
+          {%- endfor -%}
+          {{ vms }}
+        ansible_libvirt_pools: {}
+      ansible.builtin.include_role:
+        name: "libvirt_manager"
+        tasks_from: "start_vms.yml"
+        apply:
+          delegate_to: "{{ cifmw_target_host | default('localhost') }}"
+
+    # This bootstraps the controller-0 node, and RedFish virtual BMC is
+    # spawned if cifmw_use_sushy_emulator is enabled.
+    - name: Gather controller-0 facts
+      when: cifmw_use_sushy_emulator | default(true) | bool
+      delegate_to: controller-0
```

Comment:
So, with this, we are enforcing users of this playbook to have a controller-0 instance. Can we make it conditional? Like, if I need sushy -> Enforce there is controller-0

Reply:
I'm not sure I got it right, but I added conditions on:
```
        - "'controllers' in groups"
        - "'controllers' in group_names"
```
I also change the `cifmw_use_sushy_emulator` default to `false`.

Let me know, will these changes allow it to work without controller-0?